packages/protocol/src/action-protocol/element-creation.ts

```diff
@@ -14,69 +14,159 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { isArray, isString } from '../utils/typeguard-util';
-import { isAction, isActionKind, Operation } from './base-protocol';
-import { Args, Point } from './types';
+import { Point } from 'sprotty-protocol';
+import { hasArrayProp, hasStringProp } from '../utils/type-util';
+import { Operation } from './base-protocol';
+import { Args } from './types';
 
+/**
+ * Common interface for all create {@link Operation}s in GLSP.
+ * The corresponding namespace offers a helper function for type guard checks.
+ */
 export interface CreateOperation extends Operation {
+    /**
+     * The type of the element that should be created.
+     */
     elementTypeId: string;
+    /**
+     * Optional additional arguments that might be needed by the server to execute the create operation.
+     */
     args?: Args;
 }
 
-export function isCreateOperation(action: any): action is CreateOperation {
-    return isAction(action) && isString(action, 'elementTypeId');
+export namespace CreateOperation {
+    export function is(object: any): object is CreateOperation {
+        return Operation.is(object) && hasStringProp(object, 'elementTypeId');
+    }
+
+    /**
+     * Typeguard function to check wether the given object is a {@link CreateOperation} with the given `kind`.
+     * @param object The object to check.
+     * @param kind  The expected operation kind.
+     * @returns A type literal indicating wether the given object is a create operation with the given kind.
+     */
+    export function hasKind(object: any, kind: string): object is CreateOperation {
+        return CreateOperation.is(object) && object.kind === kind;
+    }
 }
+
 /**
- * In order to create a node in the model the client can send a CreateNodeOperation with the necessary information to create that node.
+ * In order to create a node in the model the client can send a CreateNodeOperation with the necessary information to create
+ * the element that corresponds to that node in the source model.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `CreateNodeOperations`.
  */
-export class CreateNodeOperation implements CreateOperation {
-    static readonly KIND = 'createNode';
-
-    constructor(
-        public readonly elementTypeId: string,
-        public location?: Point,
-        public containerId?: string,
-        public args?: Args,
-        public readonly kind: string = CreateNodeOperation.KIND
-    ) {}
-}
+export interface CreateNodeOperation extends CreateOperation {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.